lms/models/application_instance.py

```diff
@@ -121,13 +121,13 @@ class Settings(StrEnum):
             Settings.MOODLE_API_TOKEN, SettingFormat.AES_SECRET
         ),
         Settings.MOODLE_GROUPS_ENABLED: JSONSetting(
-            Settings.MOODLE_GROUPS_ENABLED, SettingFormat.BOOLEAN
+            Settings.MOODLE_GROUPS_ENABLED, SettingFormat.TRI_STATE, default=False
         ),
         Settings.MOODLE_FILES_ENABLED: JSONSetting(
-            Settings.MOODLE_FILES_ENABLED, SettingFormat.BOOLEAN
+            Settings.MOODLE_FILES_ENABLED, SettingFormat.TRI_STATE, default=False
         ),
         Settings.MOODLE_PAGES_ENABLED: JSONSetting(
-            Settings.MOODLE_PAGES_ENABLED, SettingFormat.BOOLEAN
+            Settings.MOODLE_PAGES_ENABLED, SettingFormat.TRI_STATE, default=False
```

Comment:
What is the reason for defaulting to `False` here? Consistency with the previous defaults or is there another reason?

Reply:
> I presume these features require an API key

Yes, these need an API key Falso is also the current default :+1:

Reply:
What happens if you try to enable these features without configuring an API key? Does it just fail when you try to create or launch an assignment, or do we treat the feature as disabled if the API key is missing?

Reply:
Newer services are aware of the requirements (api key, client ID etc) + the actual feature flag for their `enabled` flag. 

Older ones might be more naive.